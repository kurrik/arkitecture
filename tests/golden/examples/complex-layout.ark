# Complex layout with nested containers and arrows
root {
  direction: "horizontal"
  
  frontend {
    label: "Frontend"
    
    react {
      label: "React App"
    }
    assets {
      label: "Static Assets"
    }
  }
  
  backend {
    label: "Backend Services" 
    direction: "vertical"
    
    api {
      label: "REST API"
    }
    auth {
      label: "Auth Service"
    }
  }
  
  database {
    label: "Database"
  }
}

# Arrows showing data flow
root.frontend.react --> root.backend.api
root.backend.api --> root.database
root.backend.auth --> root.database