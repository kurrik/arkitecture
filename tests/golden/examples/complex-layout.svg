<svg xmlns="http://www.w3.org/2000/svg" width="244" height="48">
  <defs>
    <marker id="arrowhead" markerWidth="10" markerHeight="7" 
            refX="9" refY="3.5" orient="auto" markerUnits="strokeWidth">
      <polygon points="0 0, 10 3.5, 0 7" fill="black" />
    </marker>
  </defs>

  <!-- Node rectangles and labels -->
  <rect x="0" y="0" width="244" height="48" fill="white" stroke="black" stroke-width="1" />
  <rect x="0" y="0" width="96" height="48" fill="white" stroke="black" stroke-width="1" />
  <text x="48" y="24" text-anchor="middle" dominant-baseline="middle" font-family="Arial" font-size="12">Frontend</text>
  <rect x="0" y="0" width="96" height="24" fill="white" stroke="black" stroke-width="1" />
  <text x="48" y="12" text-anchor="middle" dominant-baseline="middle" font-family="Arial" font-size="12">React App</text>
  <rect x="0" y="24" width="96" height="24" fill="white" stroke="black" stroke-width="1" />
  <text x="48" y="36" text-anchor="middle" dominant-baseline="middle" font-family="Arial" font-size="12">Static Assets</text>
  <rect x="96" y="0" width="88" height="48" fill="white" stroke="black" stroke-width="1" />
  <text x="140" y="24" text-anchor="middle" dominant-baseline="middle" font-family="Arial" font-size="12">Backend Services</text>
  <rect x="96" y="0" width="88" height="24" fill="white" stroke="black" stroke-width="1" />
  <text x="140" y="12" text-anchor="middle" dominant-baseline="middle" font-family="Arial" font-size="12">REST API</text>
  <rect x="96" y="24" width="88" height="24" fill="white" stroke="black" stroke-width="1" />
  <text x="140" y="36" text-anchor="middle" dominant-baseline="middle" font-family="Arial" font-size="12">Auth Service</text>
  <rect x="184" y="0" width="60" height="48" fill="white" stroke="black" stroke-width="1" />
  <text x="214" y="24" text-anchor="middle" dominant-baseline="middle" font-family="Arial" font-size="12">Database</text>

  <!-- Arrows -->
  <line x1="48" y1="12" x2="140" y2="12" stroke="black" stroke-width="1" marker-end="url(#arrowhead)" />
  <line x1="140" y1="12" x2="214" y2="24" stroke="black" stroke-width="1" marker-end="url(#arrowhead)" />
  <line x1="140" y1="36" x2="214" y2="24" stroke="black" stroke-width="1" marker-end="url(#arrowhead)" />
</svg>