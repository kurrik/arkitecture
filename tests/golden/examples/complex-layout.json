{
  "fontSize": 12,
  "fontFamily": "Arial"
}